lib/mongoid/traversable.rb

```diff
@@ -17,6 +17,23 @@ def _parent=(p)
       @__parent = p
     end
 
+    included do 
+      def self.discriminator_key
+        @discriminator_key || Mongoid.discriminator_key
+      end
+
+      def self.discriminator_key=(value) 
+        @discriminator_key = value
+        # unless fields.has_key?(@discriminator_key)
+        #   field(@discriminator_key, default: self.name, type: String)
+        # end 
+
+        descendants.each do |child| 
+          child.discriminator_key = value
```

Comment:
Let's try replacing this with `class_attribute`.

Reply:
The problem with doing this is I can't override the methods, so I can't disallow users from setting in a child class